crates/trigger/src/loader.rs

```diff
@@ -49,18 +49,23 @@ impl Loader for TriggerLoader {
             .as_ref()
             .context("LockedComponentSource missing source field")?;
         let path = parse_file_url(source)?;
-        spin_core::Component::new(
-            engine,
-            spin_componentize::componentize_if_necessary(&fs::read(&path).await.with_context(
-                || {
-                    format!(
-                        "failed to read component source from disk at path '{}'",
-                        path.display()
-                    )
-                },
-            )?)?,
-        )
-        .with_context(|| format!("loading module {path:?}"))
+        let bytes = fs::read(&path).await.with_context(|| {
+            format!(
+                "failed to read component source from disk at path '{}'",
+                path.display()
+            )
+        })?;
+        let component = spin_componentize::componentize_if_necessary(&bytes)?;
+        let was_already_component = matches!(component, std::borrow::Cow::Borrowed(_));
+        if was_already_component {
+            println!(
+                "Warning: spin component at path {} is already a WebAssembly component. \
```

Comment:
```suggestion
                "Warning: Spin component at path {} is already a WebAssembly component. \
```

Consider also dropping `already` - it makes sense only if the user is aware that Spin wraps core modules with components (but maybe we reckon that anybody playing with CM components will know that).

Reply:
Given the heavy overloading of "component," there is a risk that readers will pick out "what is problematic" as "WebAssembly" rather than "component" (because "component" has become such a blah word that it can mean pretty much "any unit of code").  "Hmm, so it doesn't like that it's already WebAssembly?  Maybe it wants me to pass in source code...?"

Maybe we are confident that only users who are already using Spin and are consciously playing with the CM (and so realise the significance of "WebAssembly component") will ever hit this.  But if there is a way to phrase this so that it's clearly a technical term of art, a la "Wasm Component Model component" except less awkward, then that would be helpful.  "...is built to the Wasm Component Model specification" or something like that maybe?

Reply:
I've changed the warning to

> Warning: Spin component at path {} is a WebAssembly component instead of a WebAssembly module. Use of the WebAssembly component model is an experimental feature.

Reply:
Perfect - the clear contrast of component vs module hits the spot.  Thank you!